### PR TITLE
added WIP GCD stuff

### PR DESCRIPTION
**Pull request by kaonn**

### CODE REVIEW

File: library/cool/gcd.red [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+          elim (sub m (suc n)) [
+          | zero → zero
+          | suc _ → suc n
+          ]
+        | suc _ → (f (sub (suc n) (suc m')) (sub/le n m')) (suc m')
+        ]
+      ]
+    )
+
+def const/zero : nat → nat =
```
Comment (ecavallo) [inline library/cool/gcd.red]: this is a pretty confusing name!

Comment (kaonn) [reply library/cool/gcd.red]: what about always/zero?

Comment (ecavallo) [reply library/cool/gcd.red]: isn't this the identity function, though?

Comment (kaonn) [reply library/cool/gcd.red]: oops you're right. eta-expansion?

Comment (ecavallo) [reply library/cool/gcd.red]: maybe id/nat? the next definition is already named eta...

Comment (kaonn) [reply library/cool/gcd.red]: I was gonna change it to eta-equiv, but that works (commuting conversion is the right word probably)

**Comment by jonsterling** [comment]
Shall we merge it?

**Pull request merged**